commons/_shared/bs-lib-components/ui/src/components/table/BsTableColHeader.vue

```diff
@@ -1,154 +1,190 @@
-                                ></BsSearchTableCol>
-                            </q-item-section>
-                        </q-item>
-                    </q-list>
-                </q-menu>
-            </q-icon> 
-        </div>
     </div>
 </template>
 
 <script lang="ts">
-import { defineComponent, PropType } from 'vue';
-
-// import isUndefined from 'lodash/isUndefined';
-import { isUndefined } from 'lodash';
-// import isNull from 'lodash/isNull';
-import { isNull } from 'lodash';
+import { defineComponent, PropType } from "vue";
 
-
-import { QIcon, QTh, QMenu, QItem, QItemSection, QList } from 'quasar';
+import { QIcon, QTh, QMenu, QItem, QItemSection, QList } from "quasar";
 import {
-        mdiArrowUpThin,
-        mdiMagnify,
-    } from '@quasar/extras/mdi-v6';
-import BsSearchTableCol from './BsSearchTableCol.vue';
+    mdiArrowUpThin,
+    mdiMagnify,
+    mdiChevronDown,
+    mdiSortAscending,
+    mdiSortDescending,
+} from "@quasar/extras/mdi-v6";
 
 export default defineComponent({
     name: "BSTableColHeader",
     components: {
-        QIcon, QTh, QMenu, QItem, QItemSection, QList
-    ,
-    BsSearchTableCol,
-},
+        QIcon,
+        QTh,
+        QMenu,
+        QItem,
+        QItemSection,
+        QList,
+    },
     emits: ["search-col"],
     data() {
         return {
             mdiArrowUpThin,
+            sortColIcon: mdiSortAscending,
+            mdiChevronDown,
             searchColIcon: mdiMagnify,
-            searchPopupActive: false,
-            lastSearchedValue: "" as string | null | number | undefined,
             noDebounceValue: "" as string | null | number | undefined,
+            sortDesc: true,
+            sorted: false,
         };
     },
     computed: {
         sortable(): boolean {
             return !!(this.col as any)?._sortable;
         },
-        searching(): boolean {
-            const searchVal = this.lastSearchedValue;
-            return !(isNull(searchVal) || isUndefined(searchVal)) && !!(searchVal as string)?.length;
-        },
     },
     props: {
         sort: Function as PropType<(col: any) => void>,
-        col: Object as PropType<{label: string, name: string}>,
-        searchedCols: Object as PropType<Record<string, string>>
-    },
-    watch: {
-        searchPopupActive(newVal, lastVal) {
-            if (newVal || !lastVal) return;
-            this.searchColumn(this.noDebounceValue);
-        },
-        searchedCols(newVal: Record<string, string>) {
-            console.log(this.searchedCols);
-            if (this.col?.name && newVal.hasOwnProperty(this.col.name)) {
-                this.lastSearchedValue = newVal[this.col.name];
-            } else {
-                this.lastSearchedValue = "";
-            }
-        }
+        col: Object as PropType<{ label: string; name: string }>,
     },
     methods: {
         sortColumn() {
             if (this.sortable && this.sort) this.sort(this.col);
```

Comment:
- There are three states - unsorted, sorted descending, sorted ascending, you have to account for that in your boolean flag `sortDesc`.
- `sortColIcon` can be a computed property.

Reply:
<img width="1544" alt="image" src="https://github.com/dataiku/solutions-contrib/assets/36790863/d7176e20-27d9-4ff2-9364-dca554e73e72">